cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        String table = params.get("src-table").toString();
+        String sourceDb = params.get("src-db").toString();
+        String destDb= params.get("dest-db").toString();
+        String sourceStgtable=table + "_stg";
+        String bdreTechPartition=params.get("bdre-tech-pt").toString();
+        String sourceNameNodeAddress=params.get("src-nn").toString();
+        String destNameNodeAddress=params.get("dest-nn").toString();
+        String sourceJobTrackerAddress=params.get("src-jt").toString();
+        String destJobTrackerAddress=params.get("dest-jt").toString();
+        String sourceHiveConnection=params.get("src-hive").toString();
+        String destHiveConnection=params.get("dest-hive").toString();
+        Connection conn = getHiveJDBCConnection(sourceDb,sourceHiveConnection);
+        Statement st = conn.createStatement();
+        List<String> sourcePartitionList = getCurrentSourcePartitionList(st, sourceDb, table);
+
+        List<String> previousPartitionList = getLastSourcePartitionList(processId, table);
+
+        Set<String> modifiedBusinessPartitionSet = getChangedBusinessPartitionSet(sourcePartitionList, previousPartitionList);
+
+        List<String> sourceColumnList = getCurrentSourceColumnList(sourceDb, table, sourceHiveConnection);
+
+        List<String> destColumnList = getDestColumnList(processId, table);
+
+        List<String> addedColumnList = getNewRegularColumnsAtSourceList(sourceColumnList, destColumnList);
+
+        //persisting the current partition info to process logs
+        logCurrentSourcePartitions(sourcePartitionList, processId, instanceExecId, table);
+
+        //persisting the current column info to process logs
+        logCurrentSourceColumns(sourceColumnList, processId, instanceExecId, table);
+
+        //creating the source stage table
+        formStageAndDestTableDDLs(st, sourceColumnList, sourceDb, destDb, table,sourceStgtable,bdreTechPartition, processId,instanceExecId);
+
+        if(checkIfDestTableExists(destDb,table,destHiveConnection))
+            alterDestTable(st,addedColumnList,destDb,table);
+        else
+            execDestTableDDL(destDb,destHiveConnection);
+
+        filterCondition=formFilterCondition(modifiedBusinessPartitionSet,sourcePartitionColumns);
+        LOGGER.debug("filterCondition = " + filterCondition);
+
+
+        srcStgTableLocation=getTableLocation(sourceDb,sourceHiveConnection,sourceStgtable);
+        destTableLocation=getTableLocation(destDb,destHiveConnection,table);
+        migrationPreprocessorInfo=setOozieUtilProperties(sourceDb,destDb,sourceStgtable,table,sourceJobTrackerAddress,sourceNameNodeAddress,destNameNodeAddress,sourceHiveConnection,processId,instanceExecId,migrationPreprocessorInfo);
+
+        st.close();
+        conn.close();
+        return migrationPreprocessorInfo;
+    }
+
+    private List<String> getCurrentSourcePartitionList(Statement st, String sourceDb, String table) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)